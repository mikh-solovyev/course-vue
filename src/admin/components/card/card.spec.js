import card from "./card.vue";
import { mount } from "@vue/test-utils";

it("component render", async () => {
    const wrapper = mount(card);
    const component = await wrapper.find(".card-component");

    expect(component.exists()).toBe(true);
});

it("component simple slot", async () => {
    const wrapper = mount(card, {
        slots: {
            default: "Content slot"
        }
    });

    await wrapper.setProps({
        simple: true
    });

    expect(wrapper.text()).toEqual("Content slot");
});

it("snapshots", async () => {
    const wrapper = mount(card);

    expect(wrapper.element).toMatchSnapshot();
});
