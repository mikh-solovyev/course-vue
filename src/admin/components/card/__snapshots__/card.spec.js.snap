// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  class="card-component"
>
  <div
    class="header"
  >
    <div
      class="text"
    />
     
  </div>
   
  <div
    class="content"
  />
</div>
`;
